lib/v2/saraba1st/digest.js

```diff
@@ -0,0 +1,76 @@
+const got = require('@/utils/got');
+const cheerio = require('cheerio');
+const queryString = require('query-string');
+const config = require('@/config').value;
+const { parseDate } = require('@/utils/parse-date');
+const timezone = require('@/utils/timezone');
+
+module.exports = async (ctx) => {
+    const tid = ctx.params.tid;
+    const cookieString = config.saraba1st.cookie ? config.saraba1st.cookie : '';
+    const res = await got('https://bbs.saraba1st.com/2b/' + tid + '.html', {
+        headers: {
+            Cookie: cookieString,
+        },
+    });
+    const $ = cheerio.load(res.data);
+    const title = $('head title').text().replace(/-.*/, '');
+    const list = $('#threadlisttableid tbody[id^="normalthread_"] tr');
+    const count = [];
+
+    for (let i = 0; i < Math.min(list.length, 20); i++) {
+        count.push(i);
+    }
+    const resultItems = count.map(async (i) => {
+        const each = $(list[i]);
+        const floor = each.find('th.new a.s.xst').text();
+        const floorUrl = each.find('th.new a.s.xst').attr('href');
+        const contentHtml = await fetchContent('https://bbs.saraba1st.com/2b/' + floorUrl);
+        return {
+            title: `${title}:${floor}`,
+            link: new URL(floorUrl, 'https://bbs.saraba1st.com/2b/').href,
+            description: contentHtml,
+            author: each.find('td.by cite').text(),
+            pubDate: timezone(parseDate(each.find('td.by em').text()), +8),
+        };
+    });
```

Comment:
```suggestion
    const count = list
        .slice(0, ctx.query.limit ? Number(ctx.query.limit) : 20)
        .toArray()
        .map((each) => {
            each = $(each);
            const floor = each.find('th.new a.s.xst').text();
            const floorUrl = each.find('th.new a.s.xst').attr('href');
            return {
                title: `${title}:${floor}`,
                link: new URL(floorUrl, 'https://bbs.saraba1st.com/2b/').href,
                author: each.find('td.by cite').text(),
                pubDate: timezone(parseDate(each.find('td.by em').first().text()), +8),
            };
        });

    const resultItems = await Promise.all(
        count.map((i) =>
            ctx.cache.tryGet(i.link, async () => {
                i.description = await fetchContent(i.link);
                return i;
            })
        )
    );
```
`pubDate` should match the first result only, otherwise the outcome is null
You missed `Promise.all()` for `resultItems`